File: tessoku/Python/chap03/A11.py
N, X = map(int, input().split())
A = list(map(int, input().split()))

L, R = 0, N-1
ans = None
while L <= R:
    M = (L + R) // 2
    if X == A[M]:
        ans = M + 1
        break
    elif X < A[M]:
        R = M - 1
    else:
        L = M + 1

print(ans)

File: tessoku/C++/chap03/A11.cpp
#include <iostream>
using namespace std;

int N, X;
int A[100009];

int search(int x) {
    int L = 1, R = N;
    while (L <= R) {
        int M = (L + R) / 2;
        if (x < A[M]) R = M - 1;
        if (x == A[M]) return M;
        if (x > A[M]) L = M + 1;
    }
    return -1;
}

int main() {
    cin >> N >> X;
    for (int i = 1; i <= N; i++) cin >> A[i];

    int ans = search(X);
    cout << ans << endl;

    return 0;
}